core/src/main/java/org/elasticsearch/common/network/NetworkService.java

```diff
@@ -210,6 +211,7 @@ public InetAddress resolvePublishHostAddresses(String publishHosts[]) throws IOE
     }
 
     /** resolves a single host specification */
+    @SuppressForbidden(reason = "does DNS lookup when bind/publish host is specified as hostname")
```

Comment:
Maybe extract the part the must use the forbidden method into a little method so you don't have to annotate the whole thing? It'd be nice to be able to be sure that we're not doing dns lookups when we try to get the loopback address, for instance.